src/main/java/io/github/fusionflux/portalcubed/content/button/pedestal/PedestalButtonConfigScreen.java

```diff
@@ -0,0 +1,251 @@
+						display.addChild(SpacerElement.width(5), 1, 4);
+					}
+
+					display.addChild(SpacerElement.height(5), 2, 0);
+				}
+
+				pressTimeCounter.addChild(SpacerElement.height(2));
+
+				{
+					var buttons = pressTimeCounter.addChild(LinearLayout.horizontal());
+
+					buttons.addChild(pressTimeCounterButton(false));
+					buttons.addChild(pressTimeCounterButton(true));
+				}
+			}
+
+			contents.addChild(SpacerElement.width(24));
+
+			{
+				var offsetSelectButtonGrid = contents.addChild(new GridLayout());
+				offsetSelectButtonGrid.spacing(2);
+				layoutsWithTooltip.add(Pair.of(Tooltip.create(Component.translatable("container.portalcubed.pedestal_button.offset")), offsetSelectButtonGrid));
+
+				for (var offset : Offset.values()) {
+					var button = new ValueSelectButton<Offset>(
+						OFFSET_SELECT_WIDTH, OFFSET_SELECT_HEIGHT, OFFSET_SELECT_BASE,
+						offset, () -> this.offset, v -> {
+							this.offset = v;
+							konami(v);
+							this.dirty = true;
+						}, offsetSelectButtons::get
+					);
+					offsetSelectButtonGrid.addChild(button, offset.stepY + 1, offset.stepX + 1);
+					offsetSelectButtons.put(offset, button);
+				}
+			}
+		}
+
+		{
+			root.addChild(SpacerElement.height(5));
+			var footer = root.addChild(LinearLayout.horizontal());
+
+			footer.addChild(new ToggleButton(
+				BASE_TOGGLE_WIDTH, BASE_TOGGLE_HEIGHT, BASE_TOGGLE_BASE,
+				() -> base, v -> {
+					base = v;
+					dirty = true;
+				}
+			));
+			footer.addChild(new TitleWidget(Component.translatable("container.portalcubed.pedestal_button.base"), font), settings -> settings.alignVerticallyBottom().paddingLeft(2));
+		}
+
+		root.arrangeElements();
+		root.setPosition(leftPos, topPos);
+		root.visitWidgets(this::addRenderableWidget);
+		Collections.reverse(children());
+	}
+
+	@Override
+	public void render(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+		super.render(graphics, mouseX, mouseY, delta);
+		for (var layoutWithTooltip : layoutsWithTooltip) {
+			var layout = layoutWithTooltip.right();
+			boolean isHovered = mouseX >= layout.getX() && mouseY >= layout.getY() && mouseX < layout.getX() + layout.getWidth() && mouseY < layout.getY() + layout.getHeight();
+			if (isHovered)
+				setTooltipForNextRenderPass(layoutWithTooltip.left(), DefaultTooltipPositioner.INSTANCE, true);
+		}
+	}
+
+	@Override
+	public void renderBackground(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+		super.renderBackground(graphics, mouseX, mouseY, delta);
+		graphics.blit(this.style.background, this.leftPos, this.topPos, 0, 0, BACKGROUND_WIDTH, BACKGROUND_HEIGHT);
+	}
+
+	private void konami(Offset offset) {
+		konamiRecord.add(offset);
+		if (konamiRecord.equals(KONAMI_CODE)) {
+			minecraft.getSoundManager().play(SimpleSoundInstance.forUI(PortalCubedSounds.SURPRISE, 1));
+			konamiRecord.clear();
+		} else if (konamiRecord.size() >= KONAMI_CODE.size()) {
```

Comment:
This will not work right if something is input before the code

Reply:
I know, this is a change to actual function so this needs to be discussed further

Reply:
It already is?

Reply:
This is something I tested for yesterday, but also make sure that the string of inputs doesn't persist if you exit one button and open another.  it worked correctly when I tested it, but if you make changes double check that splitting the code across multiple buttons doesn't work.

Reply:
It's impossible for GUI state to carry over from one button to another

Reply:
> It already is?

it's not.
it starts empty, and each input adds one.
once the size is right, if if doesn't match, it's cleared.

You need to add the input to the start and remove the last one if the size is too large, never clear it.